File: test/cases/record/test_record_param_var.c
#include <stdio.h>
typedef struct {
    int age;
    float score;
} person;
person p;


void modifyperson(person *p);

void modifyperson(person *p) {
    (*p).age = 30;
    (*p).score = 88.0;

}


int main(void) {
    p.age = 25;
    p.score = 95.5;
    printf("%d", p.age);
    modifyperson(&p);
    printf("%d", p.age);

    return 0;
}

File: test/cases/record/test_record_param_var.pas
program test;
type
  person = record
    age: integer;
    score: real
  end;
var
  p: person;

procedure modifyPerson(var p: person);
begin
  p.age := 30;
  p.score := 88.0
end;

begin
  p.age := 25;
  p.score := 95.5;
  write(p.age);
  modifyPerson(p);
  write(p.age)
end.
